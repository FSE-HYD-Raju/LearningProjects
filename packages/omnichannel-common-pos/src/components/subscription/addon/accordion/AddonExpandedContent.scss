@import '../../../../styles/_variables.scss';

.AddonsTabView-addon-expanded-content {

	display: flex;
	flex: 1 0 0;
	flex-direction: column;
	padding: 0 2.5rem 0.5rem;
	background-color: $gray-100;
	border-top: 1px solid $weasel-border-color-solid;
	font-size: $font-size-sm;

	&-characteristics-list {
		display: flex;
		flex-direction: column;
		justify-content: flex-start;
		margin: 0.625rem -1rem;
		border-top: 1px solid $weasel-border-color-solid;
		padding: 0.5rem 1rem;
		background-color: #fff;
		font-size: $font-size-base;
	}

	&-addon-details {
		display: flex;
		flex-direction: column;
		justify-content: space-between;
		margin-top: 10px;
		margin-bottom: 10px;

		&-name {
			display: grid;
			grid-template-columns: 1fr 4fr;
			justify-content: flex-start;
		}
		&-price {
			display: flex;
			flex-direction: row;
			flex: 2;
			&-upfront {
				display: flex;
				flex: 1 0 0;
				margin-right: 10px;
			}
			&-recurrent {
				flex: 1 0 0;
				display: flex;
			}
		}
	}

	&-characteristic-list {
		flex-direction: column;
	}

	&-characteristic-list-item {
		display: flex;
		flex-direction: row;
		justify-content: space-between;
		font-size: 13px;
		padding-top: 0.2rem;
		font-size: 13px;
	}

	&-characteristic-name {
		color: gray("600");
	}

	&-characteristic-value {
		margin-left: 10px;
	}

	&-addon-description {
		font-size: 13px;
		line-height: 13px;
		white-space: pre-wrap;
		margin-right: 25px;
	}

}
