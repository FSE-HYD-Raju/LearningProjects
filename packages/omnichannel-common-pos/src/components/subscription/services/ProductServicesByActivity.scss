@import '../../../styles/_variables.scss';

.ProductServicesByActivity {
}

div[id^="OcAccordionList-"] {
	max-height: 2.8125rem;
}
