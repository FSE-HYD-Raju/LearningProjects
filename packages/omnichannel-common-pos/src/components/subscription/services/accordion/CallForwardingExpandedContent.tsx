import * as React from "react";
import addonMessages from "../../addon/Addon.messages";
import { FormattedMessage } from "../../../../channelUtils";

const CallForwardingExpandedContent: React.FC<{}> = () => {

	return (
		<div className="container AddonsTabView-addon-expanded-content">
			<div>
				<FormattedMessage {...addonMessages.callForwarding}/>
			</div>
			<div className="addon-expanded-content-addon-description">
				<FormattedMessage {...addonMessages.callForwardingDescription}/>
			</div>
		</div>
	);
};

export default CallForwardingExpandedContent;
