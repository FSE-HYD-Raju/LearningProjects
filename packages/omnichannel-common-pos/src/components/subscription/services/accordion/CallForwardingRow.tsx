import cssns from "../../../../utils/cssnsConfig";
import { Service } from "../../../../redux/types";
import { FC } from "react";
import ProductServicesUtils from "../ProductServices.utils";
import FormattedMessage from "../../../../channelUtils/FormattedMessage";
import serviceMessages from "../Services.messages";
import { CommonCustomizationPoints, withCustomization } from "../../../../customization";
import messages from "../../../../commonMessages";
import { OcButton, OcButtonSize, OcButtonType } from "../../../ocComponents";

const { React } = cssns("AddonsTabView");

interface CallForwardingRowProps {
	callForwardingServices: Array<Service>;
	onClick: () => void;
	serviceStateTransitionByActionName: Record<string, Array<string> | string>;
}

const CallForwardingRow: FC<CallForwardingRowProps> = (props: CallForwardingRowProps) => {
	const { callForwardingServices } = props;

	const lifecycleStatus = ProductServicesUtils.getCallForwardingStaus(callForwardingServices);

	return (
		<div className="list-item" id="addon-item-call-forwarding" data-customizable>
			<div className="list-item-column name">
				<FormattedMessage {...messages.servicesNameLabel} />
				<span className="call-forwarding-description">
					<FormattedMessage {...serviceMessages.configureCallForwarding}/>
				</span>
			</div>
			<div className="list-item-column fees">
				<span className="customer-subscription-services-free-addon">
					<FormattedMessage {...serviceMessages.freePrice}/>
				</span>
			</div>
			<div className="list-item-column expiration"/>
			<div className={`list-item-column status status-${lifecycleStatus}`}>
				<FormattedMessage{...serviceMessages.lifecycleStatusRow} values={{ lifecycleStatus }}/>
			</div>
			<div className="list-item-column actions">
				<OcButton
					buttonType={OcButtonType.PRIMARY}
					buttonSize={OcButtonSize.SMALL}
					id={"buttonConfigureCallforwaring"}
					key={"button-configure-call-forwaring-key"}
					onClick={props.onClick}
				>
					<FormattedMessage {...messages.configureButtonLabel} />
				</OcButton>
			</div>
		</div>
	);
};

const CallForwardingRowCustomization = withCustomization<CallForwardingRowProps>(
	CommonCustomizationPoints.CALL_FORWARDING_ROW, CallForwardingRow);

export {
	CallForwardingRowCustomization as CallForwardingRow,
	CallForwardingRow as CallForwardingRowBaseline,
	CallForwardingRowProps
};
