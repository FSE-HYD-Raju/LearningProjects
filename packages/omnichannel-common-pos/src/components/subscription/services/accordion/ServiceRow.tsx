import cssns from "../../../../utils/cssnsConfig";
const { React } = cssns("AddonsTabView");
import { LifecycleChangeAction, Service } from "../../../../redux/types";
import { ReactNode, FC } from "react";
import { get, uniqBy } from "lodash";
import ProductServicesUtils from "../ProductServices.utils";
import ServiceTransitionsMessage, { ServiceTransitionsMessageState } from "../ServiceTransitionsMessage";
import FormattedMessage from "../../../../channelUtils/FormattedMessage";
import serviceMessages from "../Services.messages";
import OcButtonWithDropdown from "../../../ocComponents/OcButtonWithDropdown";
import { CommonCustomizationPoints, withCustomization } from "../../../../customization";
import Time from "../../../../channelUtils/Time";
import { OcButton, OcButtonSize, OcButtonType } from "../../../ocComponents";

interface ServiceRowProps {
	addon: Service;
	onClick: (actions: LifecycleChangeAction) => void;
	serviceStateTransitionByActionName: Record<string, Array<string> | string>;
}

const renderActions = (buttons: Array<ReactNode>): ReactNode => {
	if (buttons && buttons.length > 0) {
		if (buttons.length > 1) {
			return (<OcButtonWithDropdown>{buttons}</OcButtonWithDropdown>);
		} else {
			return buttons;
		}
	}
	return null;
};

const ServiceRow: FC<ServiceRowProps> = (props: ServiceRowProps) => {
	const { addon } = props;
	const name = get(addon, "specification.name", "-");
	const validForEndDate = get(addon, "validFor.endDate");
	const expiryDate = validForEndDate ? Time.formatDate(validForEndDate) : "-";

	let actionButtons: Array<ReactNode> = [];
	const service = addon as Service;

	if (service.allowedTransitions && service.allowedTransitions.length > 0) {
		actionButtons = uniqBy(service.allowedTransitions
				.map((action: LifecycleChangeAction): LifecycleChangeAction => {
					return ProductServicesUtils.mapTransitions(action, props.serviceStateTransitionByActionName);
				})
				.filter((action: LifecycleChangeAction) => {
					return action.name;
				}),
			(action: LifecycleChangeAction) => action.name
		).map((action: LifecycleChangeAction): ReactNode => {
			return (
				<OcButton
					id={`buttonStateTransition-${service.id}-${action.id}`}
					key={`buttonStateTransition-${service.id}-${action.id}-key`}
					buttonType={OcButtonType.LINK}
					buttonSize={OcButtonSize.SMALL}
					onClick={() => props.onClick(action)}
				>
					<ServiceTransitionsMessage state={action.name as ServiceTransitionsMessageState}/>
				</OcButton>
			);
		});

	}

	const lifecycleStatus = ProductServicesUtils.getLifecycleStatusMessage(service.lifeCycleStatus);

	return (
		<div className="list-item" id={`addon-item-${addon.id}`} data-customizable>
			<div className="list-item-column name"><span>{name}</span></div>
			<div className="list-item-column fees">
				<span className="customer-subscription-services-free-addon">
					<FormattedMessage {...serviceMessages.freePrice}/>
				</span>
			</div>
			<div className="list-item-column expiration">{expiryDate}</div>
			<div className={`list-item-column status status-${lifecycleStatus.toLowerCase()}`}>
				<FormattedMessage {...serviceMessages.lifecycleStatusRow} values={{ lifecycleStatus }}/>
			</div>
			<div className="list-item-column actions">
				{actionButtons.length > 0 ? (
					renderActions(actionButtons)
				) : (
					<FormattedMessage {...serviceMessages.lifecycleStatusDefaultValue} />
				)}
			</div>
		</div>
	);
};

const ServiceRowCustomization = withCustomization<ServiceRowProps>(CommonCustomizationPoints.SERVICE_ROW, ServiceRow);

export {
	ServiceRowCustomization as ServiceRow,
	ServiceRow as ServiceRowBaseline,
	ServiceRowProps
};
