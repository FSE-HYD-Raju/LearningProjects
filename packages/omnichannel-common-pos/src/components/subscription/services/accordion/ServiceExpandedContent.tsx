import cssns from "../../../../utils/cssnsConfig";
import { Service } from "../../../../redux/types";
import { FC } from "react";
import { CharacteristicsContent } from "../../addon/accordion/CharacteristicsContent";

const { React } = cssns("AddonsTabView");

interface ServiceExpandedContentProps {
	addon: Service;
}

const ServiceExpandedContent: FC<ServiceExpandedContentProps> = props => {
	const { addon } = props;

	if (!addon) {
		return <span />;
	}

	return (
		<div id={"addon-expanded-content-" + addon.id} className="addon-expanded-content">
			<CharacteristicsContent characteristics={addon.characteristics}/>
		</div>
	);
};

export {
	ServiceExpandedContent,
	ServiceExpandedContentProps
};
