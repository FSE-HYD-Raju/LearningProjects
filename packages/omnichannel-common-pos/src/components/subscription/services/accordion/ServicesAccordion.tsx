import { HasId, LifecycleChangeAction, Service } from "../../../../redux/types";
import cssns from "../../../../utils/cssnsConfig";
import { OcAccordion } from "../../../ocComponents/accordion";
import { PureComponent, ReactElement } from "react";
import { ONLY_ONE_STRATEGY } from "../../../ocComponents/accordion";
import { ServiceItem } from "../../../../redux";
import { ServiceExpandedContent } from "./ServiceExpandedContent";
import { CallForwardingRow } from "./CallForwardingRow";
import CallForwardingExpandedContent from "./CallForwardingExpandedContent";
import { ServiceRow } from "./ServiceRow";
import AddonsTabTableHeaders from "../../AddonsTabTableHeaders";

const { React } = cssns("AddonsTabView");

interface ServicesAccordionProps {
	services: Array<ServiceAccordionItem>;
	serviceStateTransitionByActionName: Record<string, Array<string> | string>;
	callForwardingServices: ServiceItem | undefined;
	onServiceClick: (service: Service, action: LifecycleChangeAction) => void;
	onCallForwardingClick: () => void;
}

interface ServiceAccordionItem extends HasId {
	service?: Service;
	callForwardingServices?: Array<Service>;
}

class ServicesAccordion extends PureComponent<ServicesAccordionProps> {
	headerRendererFunction = (item: ServiceAccordionItem): {header: ReactElement<any>} => {
		let header: ReactElement<any> = <span/>;
		if (item.service) {
			header = (
				<ServiceRow
					addon={item.service}
					serviceStateTransitionByActionName={this.props.serviceStateTransitionByActionName}
					onClick={(action: LifecycleChangeAction) => this.props.onServiceClick(item.service!, action)}
				/>
			);

		} else if (item.callForwardingServices) {
			header = (
				<CallForwardingRow
					callForwardingServices={item.callForwardingServices}
					serviceStateTransitionByActionName={this.props.serviceStateTransitionByActionName}
					onClick={this.props.onCallForwardingClick}
				/>
			);
		}
		return { header: header };
	}

	contentRendererFunction = (item: ServiceAccordionItem): { content: ReactElement<any> } => {
		let content: ReactElement<any> = <span/>;
		if (item.service) {
			content = <ServiceExpandedContent addon={item.service}/>;
		} else if (item.callForwardingServices) {
			content = <CallForwardingExpandedContent/>;
		}
		return { content: content };
	}

	render() {
		return (
			<>
				<AddonsTabTableHeaders />
				<OcAccordion
					expandStrategy={ONLY_ONE_STRATEGY()}
					items={this.props.services}
					headerRendererFunction={this.headerRendererFunction}
					contentRendererFunction={this.contentRendererFunction}
				/>
			</>
		);
	}
}

export {
	ServiceAccordionItem,
	ServicesAccordion,
	ServicesAccordionProps
};
