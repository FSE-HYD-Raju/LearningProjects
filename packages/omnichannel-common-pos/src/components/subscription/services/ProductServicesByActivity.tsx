import CallForwardingConfigurationModalContainer from "./callForwarding/CallForwardingConfigurationModalContainer";
import cssns from "../../../utils/cssnsConfig";
import { Component } from "react";
import {
	CallForwardingServiceModify,
	LifecycleChangeAction,
	Product,
	Service,
	ServiceItem
} from "../../../redux/types";
import { ContextType, contextTypesValidationMap } from "../../../types";
import ServiceStateChangeModalContainer from "./ServiceStateChangeModalContainer";
import ProductServicesUtils from "./ProductServices.utils";
import { ServicesAccordion, ServiceAccordionItem } from "./accordion/ServicesAccordion";
import serviceMessages from "./Services.messages";
import ProductUtil from "../../../utils/product/ProductUtil";
import { AddonsSection } from "../AddonsSection";
import { AddonsTabLifecycleFilter } from "../AddonsTabLifecycleFilter";
import { CommonCustomizationPoints, withCustomization } from "../../../customization";
const { React } = cssns("ProductServicesByActivity");

interface ProductServicesByActivityStateProps {
	serviceStateTransitionByActionName: Record<string, Array<string> | string>;
	callForwardingServices: ServiceItem | undefined;
	callForwardingConfigurationResult?: CallForwardingServiceModify;
	showServiceStatusChangeModal: boolean;
}

interface ProductServicesByActivityOwnProps {
	agreementId: string;
	product: Product;
	lifecycleFilter?: string;
}

interface ProductServicesByActivityActionProps {
	actions: {
		resetCallForwardingConfiguration: () => void;
		initializeStateTransition: (service: Service, transition: LifecycleChangeAction) => void;
	};
}

type ProductServicesByActivityProps = ProductServicesByActivityStateProps & ProductServicesByActivityOwnProps & ProductServicesByActivityActionProps;

interface ProductServicesByActivityState {
	showStateModificationModal: boolean;
	showCallForwardingConfigurationModal: boolean;
}

class ProductServicesByActivity extends Component<ProductServicesByActivityProps, ProductServicesByActivityState> {
	static displayName = "ProductServicesByActivity";
	static contextTypes = contextTypesValidationMap;

	constructor(props: ProductServicesByActivityProps, context: ContextType) {
		super(props, context);

		this.state = {
			showStateModificationModal: false,
			showCallForwardingConfigurationModal: false
		};
	}

	componentWillReceiveProps(newProps: ProductServicesByActivityProps) {
		if (newProps.callForwardingConfigurationResult && !this.props.callForwardingConfigurationResult) {
			this.initializeCallForwardingServiceConfiguration();
		}
	}

	initializeCallForwardingServiceConfiguration = () => {
		this.props.actions.resetCallForwardingConfiguration();
		const { showCallForwardingConfigurationModal } = this.state;
		this.setState({
			showCallForwardingConfigurationModal: !showCallForwardingConfigurationModal
		});
	};

	initializeStateTransition = (service: Service, transition: LifecycleChangeAction) => {
		this.props.actions.initializeStateTransition(service, transition);
	};

	render() {
		const { product } = this.props;

		let services: Array<Service> = ProductServicesUtils.getServicesFromProduct(product);

		services = services.filter(service => service.specification && service.specification.name);

		if (this.props.lifecycleFilter && this.props.lifecycleFilter !== AddonsTabLifecycleFilter.ALL) {
			services = services
				.filter(service => service.lifeCycleStatus && service.lifeCycleStatus.toLowerCase() === this.props.lifecycleFilter!.toLowerCase());
		}

		const callForwardingServices = ProductServicesUtils.findCallforwardingServices(services, this.props.callForwardingServices);

		// filter out call forwarding services out of normal services
		if (this.props.callForwardingServices && this.props.callForwardingServices.values) {
			services = services.filter(
				service => !ProductServicesUtils.isCallforwardingService(service, this.props.callForwardingServices)
			);
		}

		const accordionItems: Array<ServiceAccordionItem> = [];
		services.forEach((service: Service) => {
			accordionItems.push({
				id: service.id,
				service: service
			});
		});

		if (callForwardingServices && callForwardingServices.length > 0) {
			accordionItems.push({
				id: ProductServicesUtils.CF_ID,
				callForwardingServices: callForwardingServices
			});
		}

		return (
			<div className="ProductServicesByActivity">
				{this.props.showServiceStatusChangeModal && (
					<ServiceStateChangeModalContainer
						requirePaymentMethodSelection={false}
						phoneNumber={ProductUtil.getPhoneNumber(this.props.product)}
						requireReasonSelect={true}
					/>
				)}
				<CallForwardingConfigurationModalContainer
					flux={this.context.flux}
					agreementId={this.props.agreementId}
					toggleModal={this.initializeCallForwardingServiceConfiguration}
					showModal={this.state.showCallForwardingConfigurationModal}
					services={callForwardingServices}
				/>
				<div className="content" id={`services-${this.props.agreementId}`}>
					<AddonsSection
						id="product-services"
						activeHeader={serviceMessages.servicesTitle}
						inactiveHeader={serviceMessages.noServicesAvailable}
						hasData={accordionItems.length > 0}
					>
						<ServicesAccordion
							services={accordionItems}
							serviceStateTransitionByActionName={this.props.serviceStateTransitionByActionName}
							callForwardingServices={this.props.callForwardingServices}
							onServiceClick={this.initializeStateTransition}
							onCallForwardingClick={this.initializeCallForwardingServiceConfiguration}
						/>
					</AddonsSection>
				</div>
			</div>
		);
	}
}

export default withCustomization(CommonCustomizationPoints.SERVICES_LIST_CONTENT, ProductServicesByActivity);
export {
	ProductServicesByActivity as ProductServicesByActivityBaseline,
	ProductServicesByActivityOwnProps,
	ProductServicesByActivityActionProps,
	ProductServicesByActivityStateProps,
	ProductServicesByActivityProps,
	ProductServicesByActivityState
};
