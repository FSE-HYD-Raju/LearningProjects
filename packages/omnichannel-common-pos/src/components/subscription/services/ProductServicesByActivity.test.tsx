import * as React from "react";
import { TestUtils, mountWithContext, shallowWithContext } from "../../../testUtils";
import juanitaAgreement from "./testData/juanita-agreement1";
import ProductServicesByActivity, { ProductServicesByActivityProps } from "./ProductServicesByActivity";
import { Product } from "../../../redux/types";
import { OcAccordionItem } from "../../ocComponents/accordion";

describe("ProductServicesByActivity", () => {
	const context = {
		flux: {
			stores: {
				UserStore: TestUtils.makeStore("context.flux.stores.UserStore", {
					user: { id: ""}
				}),
				SalesStore: TestUtils.makeStore("context.flux.stores.SalesStore", {
					products: []
				})
			}
		},
		store: TestUtils.mockReduxStore({
			service: {
				callForwardingConfigurationErrors: undefined,
				callForwardingReasonCode: ""
			}
		})
	};
	const minProps: ProductServicesByActivityProps = {
		agreementId: "juanita-agreement1-sub",
		product: {
			realizingServices: {},
			childProducts: []
		} as any as Product,
		actions: {
			initializeStateTransition: jest.fn(),
			resetCallForwardingConfiguration: jest.fn(),
		},
		serviceStateTransitionByActionName: {},
		callForwardingServices: undefined,
		showServiceStatusChangeModal: false,
	};

	it("succeeds at shallow mount without props", () => {
		const wrapper = shallowWithContext(<ProductServicesByActivity {...minProps} />, { context });
		expect(wrapper).toMatchSnapshot();
	});

	it("succeeds at deep mount without props", () => {
		mountWithContext(<ProductServicesByActivity {...minProps} />, { context });
	});

	it("renders all services, should find 8 services", () => {
		const product = juanitaAgreement.attributes.products[0];
		const wrapper = mountWithContext(
			<ProductServicesByActivity
				{...minProps}
				product={product}
				serviceStateTransitionByActionName={{
					resume: "resume",
					reactivate: "reactivate",
					disable: "disable",
					suspend: "suspend",
					deactivate: "deactivate"
				}}
			/>, { context }
		);

		const rows = wrapper.find(OcAccordionItem);
		expect(rows.length).toEqual(7);

		const cols = wrapper.find(".AddonsTabView-list-item-column");
		expect(cols.length).toBeGreaterThan(0);
		expect(cols.at(0).first().text()).toEqual("CFS_voice_1000");
		expect(cols.at(3).first().text().toLowerCase()).toEqual("active");
		expect(cols.at(4).first().text().toLowerCase()).toEqual("disable");
	});
});
