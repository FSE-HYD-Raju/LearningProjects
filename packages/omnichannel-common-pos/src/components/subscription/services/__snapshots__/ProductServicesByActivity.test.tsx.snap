// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductServicesByActivity succeeds at shallow mount without props 1`] = `
<div
  className="ProductServicesByActivity"
>
  <Connect(CallForwardingConfigurationModal)
    agreementId="juanita-agreement1-sub"
    flux={
      Object {
        "reduxStore": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "stores": Object {
          "SalesStore": Object {
            "_listeners": Array [],
            "config": Object {
              "setState": [Function],
            },
            "emitChange": [Function],
            "getState": [Function],
            "listen": [Function],
            "products": Array [],
            "setState": [Function],
          },
          "UserStore": Object {
            "_listeners": Array [],
            "config": Object {
              "setState": [Function],
            },
            "emitChange": [Function],
            "getState": [Function],
            "listen": [Function],
            "setState": [Function],
            "user": Object {
              "id": "",
            },
          },
        },
      }
    }
    services={Array []}
    showModal={false}
    toggleModal={[Function]}
  />
  <div
    className="ProductServicesByActivity-content"
    id="services-juanita-agreement1-sub"
  >
    <AddonsSection
      activeHeader={
        Object {
          "defaultMessage": "Services",
          "description": "In Service plan & products, active services, title",
          "id": "digital-life-service-plan-and-products-active-services-title",
        }
      }
      hasData={false}
      id="product-services"
      inactiveHeader={
        Object {
          "defaultMessage": "Services (none available)",
          "description": "In Service plan & products, no active services available, title",
          "id": "digital-life-service-plan-and-products-no-active-services-title",
        }
      }
    >
      <ServicesAccordion
        onCallForwardingClick={[Function]}
        onServiceClick={[Function]}
        serviceStateTransitionByActionName={Object {}}
        services={Array []}
      />
    </AddonsSection>
  </div>
</div>
`;
